torch/fx/experimental/symbolic_shapes.py

```diff
@@ -1879,9 +1889,79 @@ def remove_redundant_dynamic_results(self):
                 dynamic_results.add(dc)
         self._dynamic_results = dynamic_results
 
+    def _process_derived_dim_roots(
+        self,
+        results: Dict[str, Dict[str, Any]],
+        name_to_dim: Dict[str, Any],
+    ) -> None:
+        '''
+        With suggested fixes for derived dims, roots can be swapped,
+        e.g. dx, dx - 1 -> dy + 1, dy. Here we don't want to print out the attached name,
+        since this leads to messages like "dx - 1 = Dim(...)".
+        Instead we evaluate the new root value, and remove results for its derivations.
+        '''
+        def _check_same_range(c, dim):
+            # returns True if c & dim are both min/max ranges with same values
+            return (
+                _is_dim(dim)
+                and ("min" in c or "max" in c)
+                and dim.min == c.get("min", 0)
+                and dim.max == c.get("max", sys.maxsize - 1)
+            )
+
+        # collect roots
+        modified_roots: Set[str] = {
+            name_to_dim[k].root.__name__
+            for k in results
+            if k in name_to_dim and _is_derived_dim(name_to_dim[k])
+        }
+
+        # evaluate each root for either concrete val or min/max range
```

Comment:
The algo here is hard to follow, please add a running example. It's intricate logic, having the transformation explicitly laid out in comments, and how it's achieved with every major step of the transformation, would be good.

Also maybe add comments on why just suggesting fixes to the roots is sufficient.